stripe/src/test/java/com/stripe/android/net/PollingSyncNetworkHandlerTest.java

```diff
@@ -0,0 +1,315 @@
+package com.stripe.android.net;
+
+import com.stripe.android.exception.APIConnectionException;
+import com.stripe.android.exception.APIException;
+import com.stripe.android.exception.StripeException;
+import com.stripe.android.model.Source;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.stubbing.OngoingStubbing;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * Test class for {@link PollingSyncNetworkHandler}.
+ */
+public class PollingSyncNetworkHandlerTest {
+
+    private static final String DUMMY_SOURCE_ID = "sourceId";
+    private static final String DUMMY_CLIENT_SECRET = "clientSecret";
+    private static final String DUMMY_PUBLISHABLE_KEY = "pubKey";
+
+    @Mock Source mCancelledSource;
+    @Mock Source mChargeableSource;
+    @Mock Source mConsumedSource;
+    @Mock Source mFailedSource;
+    @Mock Source mPendingSource;
+
+    @Mock SourceRetriever mSourceRetriever;
+
+    private PollingParameters mPollingParameters;
+    private PollingSyncNetworkHandler.TimeRetriever mTimeRetriever;
+    private PollingSyncNetworkHandler mPollingSyncNetworkHandler;
+
+    @Before
+    public void setup() {
+        MockitoAnnotations.initMocks(this);
+
+        mTimeRetriever = new PollingSyncNetworkHandler.TimeRetriever() {
+            @Override
+            public long getCurrentTimeInMillis() {
+                return System.currentTimeMillis();
+            }
+        };
+        when(mConsumedSource.getStatus()).thenReturn(Source.CONSUMED);
+        when(mChargeableSource.getStatus()).thenReturn(Source.CHARGEABLE);
+        when(mPendingSource.getStatus()).thenReturn(Source.PENDING);
+        when(mCancelledSource.getStatus()).thenReturn(Source.CANCELED);
+        when(mFailedSource.getStatus()).thenReturn(Source.FAILED);
+
+        mPollingParameters = new PollingParameters(
+                10000L,
+                100L,
```

Comment:
Using a shorter timeout for testing because this test doesn't move the clock forward programatically. If I put it in as 1000L for the initial delay, then some of the longer tests would take as much as 15 seconds to run.